gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java

```diff
@@ -381,6 +383,15 @@ protected String defaultProject() {
     return projectId != null ? projectId : googleCloudProjectId();
   }
 
+  private static String activeGoogleCloudConfig(File configDir) {
+    try {
+      return Files.readFirstLine(new File(configDir, "active_config"), Charset.defaultCharset());
```

Comment:
Are you saying that we need no worry about the case of an empty file?
Clearly file could be modified/truncated by users (though maybe that is not much of a concern?)
Personally I would prefer to be slightly more resilient in that case (and fallback to config_default rather than config_null) however it is a Nit and feel free to merge if you think we should not care about it.